colcon_bundle/verb/_archive_generators.py

```diff
@@ -0,0 +1,174 @@
+import os
+import shutil
+import tarfile
+
+from colcon_bundle.verb import logger
+from colcon_bundle.verb.bundlefile import Bundle
+
+
+def generate_archive_v1(path_context):
+    """
+    Generate bundle archive.
+
+    output.tar.gz
+    |- version
+    |- metadata.tar
+    |- bundle.tar
+
+    :param path_context: PathContext object including path configurations
+    """
+    # install_base: Directory with built artifacts from the workspace
+    install_base = path_context.install_base()
+    # staging_path: Directory where all dependencies have been installed
+    staging_path = path_context.staging_path()
+
+    logger.info('Copying {} into bundle...'.format(install_base))
+    assets_directory = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), 'assets')
+    shellscript_path = os.path.join(assets_directory, 'v1_setup.sh')
+    shutil.copy2(shellscript_path, os.path.join(staging_path, 'setup.sh'))
+    bundle_workspace_install_path = os.path.join(
+        staging_path, 'opt', 'install')
+    if os.path.exists(bundle_workspace_install_path):
+        shutil.rmtree(bundle_workspace_install_path)
+    shutil.copytree(install_base, bundle_workspace_install_path)
+
+    logger.info('Archiving the bundle output')
+    print('Creating bundle archive...')
+
+    bundle_tar_path = path_context.bundle_tar_path()
+    metadata_tar_path = path_context.metadata_tar_path()
+    archive_tar_gz_path = path_context.archive_tar_gz_path()
+
+    with tarfile.open(metadata_tar_path, 'w') as archive:
+        archive.add(path_context.installer_metadata_path(),
+                    arcname='installers.json')
+
+    if os.path.exists(bundle_tar_path):
+        os.remove(bundle_tar_path)
+
+    _recursive_tar_in_path(bundle_tar_path, staging_path)
+
+    version_file_path = path_context.version_file_path()
+    with open(version_file_path, 'w') as v:
+        v.write('1')
+
+    with tarfile.open(
+            archive_tar_gz_path, 'w:gz', compresslevel=5) as archive:
```

Comment:
why 5? That would require a comment, and some data to prove this is usually the correct parameter.